File: frauddetection/src/main/java/spendreport/dtos/CardTransaction.java
import com.fasterxml.jackson.annotation.JsonProperty;

@Data
public class CardTransaction {
    @JsonProperty("amount")
    private int amount;
    @JsonProperty("limit_left")
    private int limitLeft;
    @JsonProperty("currency")
    private String currency;
    @JsonProperty("latitude")
    private double latitude;
    @JsonProperty("longitude")
    private double longitude;
    @JsonProperty("card")
    private Card card;
    @JsonProperty("owner")
    private CardOwner owner;
    @JsonProperty("utc")
    private Date utc;
} 

File: dashboard/src/lib/cardTransaction.ts
import { Card } from "./card";
import { CardOwner } from "./cardOwner";

export class CardTransaction {
    amount: number;
    limitLeft: number;
    currency: string;
    latitude: number;
    longitude: number;
    card: Card;
    owner: CardOwner;
    utc: Date;
  
    constructor({ amount, limit_left, currency, latitude, longitude, card, owner, utc }: { amount: number; limit_left: number; currency: string; latitude: number; longitude: number; card: any; owner: any; utc: string }) {
        this.amount = amount;
        this.limitLeft = limit_left;
        this.currency = currency;
        this.latitude = latitude;
        this.longitude = longitude;
        this.card = new Card(card);
        this.owner = new CardOwner(owner);
        this.utc = new Date(utc);
    }
}